### PR TITLE
dotnet: add livecheck

### PR DESCRIPTION
**Pull request by Bo98**
The version we want to track here is a little complicated: https://github.com/dotnet/source-build/#support

> NET Source-Build is supported on the oldest available .NET SDK feature update. For example, if both .NET 6.0.1XX and 6.0.2XX feature updates are available from [dotnet.microsoft.com](https://dotnet.microsoft.com/en-us/download/dotnet/6.0), Source-Build will support 6.0.1XX.

The livecheck attempts to follow that.

### CODE REVIEW

**Review by samford — COMMENTED** [review]
I've gone through the `livecheck` block myself to confirm the behavior and this looks good to me. Nice work!

I'm about to go to sleep over here but if someone gets to this before I'm conscious again: This is fine to approve/merge after the minor suggestion below is addressed. If we don't agree on the suggested change, we can just merge this as-is.

File: Formula/dotnet.rb
Comment (Bo98) [reply Formula/dotnet.rb:18]: Sounds good to me. :+1:

**Review by samford — APPROVED** [review]
Thanks, Bo!

**Pull request merged**
